### ISSUE TITLE
`@JsonIgnoreProperties` does not stack

### ISSUE THREAD

**Issue by markwoon** [FasterXML/jackson-databind#1037]
I'd like to be able to aggregate `@JsonIgnoreProperties`, but it looks like it's only accepting the "top" one.

Here's a test case:

```
  @JsonIgnoreProperties(value = { "generated" }, allowGetters = true)
  static class BaseBean {
    public String getGenerated() {
      return "http://bar.com";
    }
  }

  @JsonIgnoreProperties(value = { "computed" }, allowGetters = true)
  static class ReadOnlyBean extends BaseBean {
    public int getComputed() {
      return 32;
    }
  }

  @Test
  public void testReadOnlyProp() throws Exception {

    ObjectMapper m = new ObjectMapper();
    String json = m.writeValueAsString(new ReadOnlyBean());
    assertTrue(json.contains("generated"));
    assertTrue(json.contains("computed"));
    ReadOnlyBean bean = m.readValue(json, ReadOnlyBean.class);
    assertNotNull(bean);
  }
```

When the `@JsonIgnoreProperties` is on a mixin, the mixin's annotation trumps all.

Would it be possible to aggregate all the `@JsonIgnoreProperties`?

**Comment by cowtowncoder** [comment]
Correct, this is the current behavior, with the exception that if Class has annotation as well as property with value of that type, then both sets are combined (union is used).

This is not an optimal behavior, but it is driven by the way all annotations are handled, with respect to inheritance: since annotation handler (which also weaves in mix-ins) has no knowledge of semantics of particular annotations (nor is there a way to actually instantiate Annotation types even if it did), there is simply no way to merge annotations from super-types.

This is not to say this could never be done, but it would require significant improvements to annotation handling. I have actually thought about this, and one particular example was indeed `@JsonIgnoreProperties`.

So I will keep this issue open and I hope we can figure out the way to get this done in future; along with a marker that indicates alternative of full override (current behavior) as well.

Thank you for suggesting this.

**Comment by huehnerlady** [comment]
Hi, is there any update on this? Or can I vote on this change?

**Comment by cowtowncoder** [comment]
@huehnerlady No -- there is no invisible magical update that just happened to be done without updating issue. We don't do that.

As to voting: if github has a mechanism feel free to. The problem here unfortunately that of implementation: to make this happen, the whole annotation processing system would need to be rewritten as per earlier comment. I hope this can be done for 3.0, but it will be ways off.

**Comment by huehnerlady** [comment]
@cowtowncoder thank you for your quick answer, I will keep watching this issue then and hopefully this will be done in t=3.0 :)

**Comment by JacksonJang** [comment]
@markwoon @huehnerlady 
I've checked it just now, and it looks like this issue has already been resolved.

**Comment by cowtowncoder** [comment]
To be honest, I am curious -- there is no merging of annotations as suggested, so maybe there is something about test that is wrong.

That is, there's no fix.

Looking at test, I don't see how it would fail, either; I think the reason it passes is actually because 3.0 defaults `DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES` to `false`.
So: "allowGetters = true" means ignoral only affects setters.

So I think test here is broken, but stacking does not happen.

**Comment by cowtowncoder** [comment]
Oh.... now I see. I think I get what test tries to do -- it leaves getters (so serialization works). But tries to block setters.

So, enabling `DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES` would make test fail as expected (and like it does in 2.0)

**Comment by JacksonJang** [comment]
Ah… once I enabled `DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES`,
I was finally able to detect the issue.

Unfortunately, there is still a problem.
I'll work on fixing this part and follow up with an update.

**Comment by JacksonJang** [comment]
@cowtowncoder 
I've just pushed!
Could you please take a look when you have a moment?

**Comment by cowtowncoder** [comment]
@JacksonJang while I appreciate your proposing solution, in this case I don't think that approach should be used.

We can still add failing test of course.
